requests_oauthlib/oauth2_session.py

```diff
@@ -97,6 +93,11 @@ def __init__(
             "protected_request": set(),
         }
 
+    def auth(self, r):
```

Comment:
Could the same have been done with the lambda function?

Reply:
That's how it is in the original code, but it makes the OAuth2 session not picklable...